src/config.ts

```diff
@@ -0,0 +1,46 @@
+import * as dotenv from 'dotenv';
+import {
+	getArray, getRecord,
+	getRequiredArray,
+	getRequiredNumber,
+	getRequiredString,
+	getSchemaDef,
+	getUrl
+} from "./config-validator.js";
+
+if (dotenv.config().error) {
+	console.log(`Error loading environment variables, aborting.`);
+	process.exit(1);
+}
+
+
+export const dbHost = getRequiredString('DB_HOST');
+export const dbUsername = getRequiredString('DB_USERNAME');
+export const dbPassword = getRequiredString('DB_PASSWORD');
+export const dbSchema = getRequiredString('DB_SCHEMA');
+export const dbName = getRequiredString('DB_NAME');
+export const requestRoute = getRequiredString('REQUEST_ROUTE');
+
+export const egoUrl = getUrl('EGO_URL');
+
+export const dbPort = getRequiredNumber('DB_PORT');
+export const port = getRequiredNumber('PORT');
+
+export const dbSync: boolean = JSON.parse(process.env.DB_SYNCHRONIZE || 'false');
+export const logging: boolean = JSON.parse(process.env.DB_LOGGING || 'false');
+
+export const entityList = getRequiredArray('ENTITY_LIST');
+export const scopes = getRequiredArray('SCOPES');
+
+export const dbSequences = getArray('DB_SEQUENCES');
+
+
+entityList.forEach((entity) => {
+	getSchemaDef(entity).parse(JSON.parse(process.env[entity.toUpperCase() + `_SCHEMA`] || '[]'));
+});
+
+
+entityList.forEach((entity) => {
+	getRecord(entity).parse(JSON.parse(process.env[entity.toUpperCase() + `_SEARCH`] || '[]'));
+});
```

Comment:
Are these two loops just validating the evnironment variables? We are we not also parsing them into config variables?

For example, we use the schemas in `getTableDefinition()`, where we have to repeat the parsing process once more.

Perhaps we should create a map of all our `tableDefinitions` like so:

```ts
export const tableDefinitions = entityList.reduce<Map<string, SchemaInfo>>((acc, entityName) => {
	const tableDefinition = getSchemaDef(entityName);
	return acc.set(entityName, tableDefinition);
}, new Map());
```

And then each table definition can be referenced from this map in the future.

Reply:
I'm not sure what the `getRecord` loop is used for. Can you help me understand that?